retrieval/discovery/kubernetes/discovery.go

```diff
@@ -110,13 +104,9 @@ type Discovery struct {
 
 	apiServers   []config.URL
 	apiServersMu sync.RWMutex
-	nodes        map[string]*Node
 	services     map[string]map[string]*Service
```

Comment:
If we're breaking out by Artifact type, shouldn't services also be dropped? (also line 118/109, for servicesMu)